tools/config/config.go

```diff
@@ -0,0 +1,321 @@
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	"k8s.io/client-go/util/connrotation"
+	"k8s.io/klog/v2"
+	"net"
+	"net/http"
+	"reflect"
+	"sigs.k8s.io/cluster-inventory-api/apis/v1alpha1"
+	"sigs.k8s.io/cluster-inventory-api/tools/plugins"
+	"sigs.k8s.io/cluster-inventory-api/tools/plugins/spiffe"
+	"sync"
+	"time"
+)
+
+const (
+	APIServerEndpointPropertyKey = "cluster-entrypoints.k8s.io"
+	CABunblePropertyKey          = "cabundle.k8s.io"
+	PluginPropertyKey            = "plugin.multicluster.k8s.io"
+)
+
+var pluginMap = map[string]plugins.PluginInterface{}
```

Comment:
Hi qiujian16! Since this is not exposed, for exec plugins I assume people wouldn't be able to use the `BuildConfigFromClusterProfile`, if I understand it correctly?